### PR TITLE
Fix: handle VCD variable references with and without whitespace

### PR DESCRIPTION
**Pull request by RCoeurjoly**
_What are the reasons/motivation for this change?_

Fixes https://github.com/YosysHQ/yosys/issues/4617.

_Explain how this is achieved._

We detect if a variable reference in the VCD file contains a space between the signal name and the bit range. If no space is detected, the bit or part-select is parsed and removed from the signal name.

_If applicable, please suggest to reviewers how they can test the change._

In addition to the added test case, I processed the [neorv32](https://github.com/stnolting/neorv32) design with its GHDL-generated VCD file, and the `sim` pass did not report any issues. It might be helpful to test with other real-world examples, but I believe the current testing is sufficient to verify the fix

### CODE REVIEW

**Review by mmicko — APPROVED** [review]
Just needed to be rebased on vcd2fst integration when that one is merged. 
addr variable could use a better name as well.

Would wait till after the release before merging this, so we can try out more designs and check for regressions

**Comment by RCoeurjoly** [comment]
> Just needed to be rebased on vcd2fst integration when that one is merged. addr variable could use a better name as well.
> 
> Would wait till after the release before merging this, so we can try out more designs and check for regressions

Rebased on main. Changed `addr` for `index_or_range`.

**Pull request merged**
